azure-iot-hub/tests/test_iothub_configuration_manager.py

```diff
@@ -47,10 +49,115 @@ def iothub_configuration_manager():
         skn=fake_shared_access_key_name,
         sk=fake_shared_access_key,
     )
-    iothub_configuration_manager = IoTHubConfigurationManager(connection_string)
+    iothub_configuration_manager = IoTHubConfigurationManager.from_connection_string(
+        connection_string
+    )
     return iothub_configuration_manager
 
 
+@pytest.mark.describe("IoTHubConfigurationManager - .from_connection_string()")
+class TestFromConnectionString(object):
+    @pytest.mark.parametrize(
+        "connection_string",
+        [
+            "HostName={hostname};DeviceId={device_id};SharedAccessKeyName={skn};SharedAccessKey={sk}".format(
```

Comment:
Use the pytest.param() syntax here so you can provide an id for each test case that will provide readable output (Just l